model_evaluation/input_fn_example.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Defines some examples of input_fn for the evaluation notebook."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import numpy as np
+import pandas as pd
+import pkg_resources
+import os
+import random
+import re
+
+import tensorflow as tf
+from tensorflow.python.lib.io import file_io
+
+from unintended_ml_bias import model_bias_analysis
+from utils_export import utils_tfrecords
+
+
+#Faster to access GCS file + https://github.com/tensorflow/tensorflow/issues/15530
+os.environ['GCS_READ_CACHE_MAX_SIZE_MB'] = '0'
+
+
+#TODO(fprost): Clean this file.
+
+#### #### #### #### #### ####
+#### PERFORMANCE DATASET ####
+#### #### #### #### #### ####
+
+
+def create_input_fn_toxicity_performance(tokenizer, model_input_comment_field):
```

Comment:
Thanks for the suggestion, this file is indeed quite messy and any ideas are welcomed.

The problem that I have here is that input_fn implementation are very dependent on your test file and have a lot of variations:
- Format: civil/toxicity datasets are tfrecords (in this case, utils_tfrecords.decode_tf_records_to_pandas does most of the work) but the synthetic one is pandas.
- Fields to parse: list of identity columns.
- Transformation to make on each field: labels might have to be turned into boolean.
- Filtering, we might want to keep a subset of the examples (e.g. keep only the ones with identity or apply a random filter).
- Name of the fields can change
+ anything that another test file might have (e.g. multiple labels)

Any idea how to make it more general?

Reply:
Yeah that makes sense. The only thing I can think of is adding a separate base function for each type of data (tfrecord vs pandas) and adding an arg that accepts a transformation function that will convert labels into the appropriate type. Perhaps just easier to have a separate one per dataset then, even if it means some repeated code. From a user standpoint it may not matter too much as long as it's clear in the documentation which function needs to be called.

Reply:
Your suggestion makes sense and I agree it could be nicer.

Can I leave it as a TODO for now and we will revisit it when we have an idea if it will be common to write new/custom input_fn? Or do we want to make it a requirement for this PR?

Reply:
Also, to your point, from docstring+ naming, is it clear which input_fn to use?

Reply:
Yup, I think it makes sense to revisit this later, the docs make it clear which one to use.